observIQ/bindplane-agent
# feat: Update to v2

### Proposed Change


Reworks updater to update an installation from v1 to v2.
- Reads `endpoint`, `secret_key`, and `agent_id` values from existing manager.yaml and uses them to create basic supervisor.yaml
- Uses healthcheck extension on agent to determine if startup is successful once service is restarted
- Cleans up various files from v1 install that are no longer needed in a v2 install

Note:
- This does not apply the previously used config, users will need to reapply a config from BP
- Only the values listed above are carried over from the `manager.yaml`

### Testing
Test following this guide: https://docs.google.com/document/d/1ZaSXnJIu-rjctiihBdBLvHymuaB7GFIW8I56biN5v8Q/edit?tab=t.0

1. Run `VERSION=v2.0.0 make release-test` in this repo
2. Run the following script as `./build-agent-upgrade.sh version=v2.0.0` from the root of this repo:
```bash
#!/bin/sh
VERSION="$1"

ref_name=$(git rev-parse HEAD | cut -b 1-8)

mkdir -p ./artifacts
mkdir -p ./tmp
cp ./scripts/install/*.sh ./artifacts
cp ./dist/*tar.gz ./artifacts
cp ./dist/*.rpm ./artifacts
cp ./dist/*.deb ./artifacts
cp ./dist/*SHA256SUMS ./artifacts
tar -czvf "./tmp/observiq-otel-collector-${VERSION}-SNAPSHOT-${ref_name}-artifacts.tar.gz" -C ./artifacts .

rm -rf ./artifacts
```
3. Make sure you have a v1.64.0 or older Agent running and connected to local BP.
4. Restart local BP instance after changing/adding top level config key `offline: true`.
5. Apply the agent-upgrade to BP using `bindplane upload agent-upgrade ~/bindplane-agent/tmp/observiq-otel-collector-...`.
6. Select the agent in the BP agent table and select 'Upgrade Outdated Agent'.
7. Verify Agent shuts down and comes back as a v2 agent with the same ID and is running.

##### Checklist
